client/app/states/services/custom_button_details/custom_button_details.state.spec.js

```diff
@@ -159,48 +161,84 @@ describe('State: services.custom_button_details', () => {
       }
     }
 
-    beforeEach(() => {
-      bard.inject('$controller', '$log', '$state', '$stateParams', '$rootScope', 'CollectionsApi', 'Notifications', 'DialogFieldRefresh')
-      sinon.stub(DialogFieldRefresh, 'refreshDialogField')
-      const dialogResponse = {content: [dialog], id: 213}
-      const apiQueries = sinon.stub(CollectionsApi, 'get')
-      apiQueries.onFirstCall().returns(Promise.resolve(dialogResponse))
-      apiQueries.onSecondCall().returns(Promise.resolve({}))
+    const dialogResponse = {
+      content: [dialog],
+      id: 213
+    }
+
+    it('sends target_type=vm service dialog query on init', (done) => {
```

Comment:
I think we're still missing the spec for when the `vmId` doesn't exist, right? There are still cases where the service dialogs endpoint is going to request with the `target_type` of `'service'` and the `target_id` with that service's id.

I guess maybe this spec didn't exist before, but I think it needs to be added now that we have these two different calls happening.

Reply:
Good catch :+1:. The tests should probably be reorganized a bit to be more symmetrical, and looks like we have some SUI test inconsistencies (3 tests randomly in jest not karma), can't use object spread there, etc.

But for now, I just added a similar test for init *without* `vmId` :).